src/config.ts

```diff
@@ -1,16 +1,55 @@
 import { config } from 'dotenv';
+import { ethers } from 'ethers';
 
 // Load environment variables
 config();
 
-if (!process.env.SUPABASE_URL || !process.env.SUPABASE_KEY) {
-  throw new Error('Missing Supabase credentials in environment variables');
+// Validate required environment variables
+const requiredEnvVars = [
+  'RPC_URL',
+  'STAKER_CONTRACT_ADDRESS',
+  'SUPABASE_URL',
+  'SUPABASE_KEY',
```

Comment:
As noted separately, these probably shouldn't be required fields, if we want to support json as an option (which I think we should. But ideally if the user selected `supabase` but didn't provide these, then we would fail.